From 1ee9fe190c94aec0a3122f805611b2f8c967ff23 Mon Sep 17 00:00:00 2001
From: Benedikt Spranger <b.spranger@linutronix.de>
Date: Wed, 6 Dec 2023 15:08:50 +0100
Subject: [PATCH] Retry download on failures

debootstrap failed rather sporadic due to network failures at our site.
A common debootstrap log contains log passages like below:

...
[2023-12-05T18:42:28.605Z] I: Retrieving vim-common 2:9.0.1378-2
[2023-12-05T18:42:28.605Z] W: Couldn't download package vim-common (ver 2:9.0.1378-2 arch all) at http://deb.debian.org/debian/pool/main/v/vim/vim-common_9.0.1378-2_all.deb
[2023-12-05T18:42:28.605Z] I: Retrieving vim-tiny 2:9.0.1378-2
[2023-12-05T18:42:28.868Z] W: Couldn't download package vim-tiny (ver 2:9.0.1378-2 arch amd64) at http://deb.debian.org/debian/pool/main/v/vim/vim-tiny_9.0.1378-2_amd64.debProject build was not successful, current status: build_failed
[2023-12-05T18:42:28.868Z]
[2023-12-05T18:42:28.868Z] I: Retrieving whiptail 0.52.23-1+b1
[2023-12-05T18:42:28.868Z] W: Couldn't download package whiptail (ver 0.52.23-1+b1 arch amd64) at http://deb.debian.org/debian/pool/main/n/newt/whiptail_0.52.23-1+b1_amd64.deb
[2023-12-05T18:42:28.868Z] I: Retrieving zlib1g 1:1.2.13.dfsg-1
[2023-12-05T18:42:28.868Z] W: Couldn't download package zlib1g (ver 1:1.2.13.dfsg-1 arch amd64) at http://deb.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.13.dfsg-1_amd64.deb
[2023-12-05T18:42:28.869Z] E: Couldn't download packages: libtasn1-6 libtext-charwidth-perl libtext-iconv-perl libtext-wrapi18n-perl libtinfo6 libtirpc-common libtirpc3 libudev1 libunistring2 libuuid1 libxtables12 libxxhash0 libzstd1 login logrotate logsave mawk mount nano ncurses-base ncurses-bin netbase nftables passwd perl-base procps readline-common sed sensible-utils systemd systemd-sysv sysvinit-utils tar tasksel tasksel-data tzdata udev usr-is-merged util-linux util-linux-extra vim-common vim-tiny whiptail zlib1g
...

commit 733069bb97bd ("Retry corrupted downloads rather than carrying on almost regardless. Patch mostly due to Michael Gilbert, rearranged somewhat by me (closes: #618920).") stated to "retry corrupted downloads rather than
carrying on almost regardless."

But instead of continuing the inner retry loop on failure it continues
the outer loop aka the next file.

Continue the inner retry loop.

Fixes: 733069bb97bdfe3f9c16ca4c9ef58685205eabf3

Reported-by: Hartmut Eilers <eilers@linutronix.de>
Signed-off-by: Benedikt Spranger <b.spranger@linutronix.de>
---
 functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/functions b/functions
index 00c82fa..68a6ab1 100644
--- a/functions
+++ b/functions
@@ -410,7 +410,7 @@ get () {
 			fi
 			if [ ! -e "$dest2" ]; then
 				if [ -z "$from2" ] || ! just_get "$from2" "$dest2"; then
-					if ! just_get "$from" "$dest2"; then continue 2; fi
+					if ! just_get "$from" "$dest2"; then continue; fi
 				fi
 			fi
 			if [ "$checksum" != "" ]; then
-- 
2.42.0

